README.md

```diff
@@ -57,48 +60,253 @@ This is work in progress as of 14 Nov 2018:
+
 ```
 npx @airtasker/spot init
 ```
 
 You can then run a generator with:
+
 ```
 npx @airtasker/spot generate --api api.ts
 ```
 
+## `@api`
+
+Define an API. This is required and must only be defined once:
+
+```TypeScript
+import { api } from "@airtasker/spot";
+
+@api({
+  name: "My API",
+  description: "My really cool API"
+})
+class MyAPI {}
+```
+
+| Field         | Description                           |
+| ------------- | ------------------------------------- |
+| `name`        | (**required**) Name of the API        |
+| `description` | (**required**) Description of the API |
+
+## `@endpoint`
+
+Define a HTTP endpoint for the API. An endpoint describes a particular HTTP action on a URL path:
+
+```TypeScript
+import { endpoint, genericError, header, pathParam, queryParam, request, response, specificError } from "@airtasker/spot";
+
+class MyUserEndpoints {
+  @endpoint({
+    method: "GET",
+    path: "/users",
+    description: "Retrieve all users",
+    tags: ["Users"]
+  })
+  getUsers(@queryParam({ description: "Search term" }) search_term: string): UserResponse[] {
+    return response();
+  }
+
+  @endpoint({
+    method: "GET",
+    path: "/users/:id",
+    description: "Get user by id",
+    tags: ["Users"]
+  })
+  getUser(@pathParam({ description: "Unique user identifier" }) id: string): UserResponse {
+    return response();
+  }
+
+  @endpoint({
+    method: "POST",
+    path: "/users",
+    description: "Create a user",
+    tags: ["Users"]
+  })
+  @specificError<ApiError>({
+    name: "unauthorized",
+    statusCode: 401
+  })
+  createUser(
+    @request req: CreateUserRequest,
+    @header({
+      name: "Authorization",
+      description: "This is the authorization token"
+    })
+    authToken: Optional<string>
+  ): CreateUserResponse {
+    return response();
+  }
+}
+
+interface UserResponse {
+  firstName: string;
+  lastName: string;
+}
+
+interface CreateUserRequest {
+  firstName: string;
+  lastName: string;
+}
+
+interface CreateUserResponse {
+  success: boolean;
+}
+
+interface ApiError {
+  message: string;
+}
+```
+
+| Field                | Default            | Description                                               |
+| -------------------- | ------------------ | --------------------------------------------------------- |
+| `method`             |                    | (**required**) [HTTP method](#suppported-http-methods)    |
+| `path`               |                    | (**required**) URL path                                   |
+| `description`        | `""`               | Description of the endpoint                               |
+| `requestContentType` | `application/json` | Content type of the request body                          |
+| `successStatusCode`  | `200`              | HTTP status code for a successful response                |
+| `tags`               |                    | Array of tags used for endpoint grouping in documentation |
```

Comment:
Nit: I personally find the term `List` less implementation-specific than `Array` (to me, "array" implies that indexes are important). Thoughts?

Reply:
Makes perfect sense